packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/ProxyMethod.java

```diff
@@ -363,6 +363,10 @@ private IType mapToSyncType(IType type) {
                             JavaSettings.getInstance().isInputStreamForBinary()
                                 ? ClassType.INPUT_STREAM
                                 : ClassType.BINARY_DATA);
+                    } else if (innerGenericType.getName().equals("Response")
+                        && innerGenericType.getTypeArguments()[0] == GenericType.FLUX_BYTE_BUFFER
+                        && JavaSettings.getInstance().isFluent()) {
```

Comment:
Wondering why the mapping from async `Response<Flux<ByteBuffer>>` to sync `Response<BinaryData>` happens only for `fluent` case, but generally for all sync

Reply:
Here: https://github.com/microsoft/typespec/blob/7011df1f9822ad95b943224ff6abf9de9384d892/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ResponseTypeFactory.java#L43-L48

The logic to map response type to FluxBB includes fluent check, hence the check here. 

Though IMHO the method name `isNotPagingOperation` is not appropriate for its purpose. It should be `isNotNextPageOperation` and I've updated the method name. Let me know if any concern.

Reply:
The reason is that fluent is done much earlier than the introduction of `BinaryData` class in azure-core.

Hence, at the start, there is only `Flux<ByteBuffer>` to choose in async API. If we had `BinaryData` at that time, we probably would do `Mono<BinaryData>`.

PS: binary, because there is handling in LRO strategy on the response, and binary is easier to process than a model class instance.

Reply:
thanks for the detailed context. The new method name `isNotNextPageOperation` makes it more clear!